From 05c5f92cf7d6660d2829675b2d55af124e1f4d57 Mon Sep 17 00:00:00 2001
From: Katsuya Matsubara <matsu@igel.co.jp>
Date: Wed, 17 Oct 2012 18:05:53 +0900
Subject: [PATCH 73/97] Revert "ext/dfbvideosink: adjust alignment of output
 image's width for MERAM"

This reverts commit ef2898e0e59581953f26cc7a3b67fb7b8076ee0c.

Conflicts:

	ext/directfb/dfbvideosink.c
---
 ext/directfb/dfbvideosink.c |   26 --------------------------
 1 file changed, 26 deletions(-)

diff --git a/ext/directfb/dfbvideosink.c b/ext/directfb/dfbvideosink.c
index fe7ab07..793765f 100644
--- a/ext/directfb/dfbvideosink.c
+++ b/ext/directfb/dfbvideosink.c
@@ -2220,9 +2220,6 @@ gst_dfbvideosink_show_frame (GstBaseSink * bsink, GstBuffer * buf)
 #if defined(HAVE_SHVIO)
     DFBSurfacePixelFormat src_format, dst_format;
     guint8 *src_datay, *src_datac;
-#if defined(HAVE_SHMERAM)
-    gint surplus;
-#endif
 #endif
 
     /* As we are not blitting no acceleration is possible. If the surface is
@@ -2259,31 +2256,8 @@ gst_dfbvideosink_show_frame (GstBaseSink * bsink, GstBuffer * buf)
       ret = GST_FLOW_UNEXPECTED;
       goto beach;
     }
-
     gst_dfbvideosink_center_rect (src, dfbvideosink->window, &result, TRUE,
         dfbvideosink->keep_ar);
-#if defined(HAVE_SHMERAM)
-    /* adjust the destination width
-       since MERAM imposes 16-byte alignment on it */
-    surplus = result.w % byte2pixel (16, dst_format);
-    if (surplus > 0) {
-      if (dfbvideosink->keep_ar) {
-        gint new_w, new_h;
-
-        /* tweak scaling factor */
-        new_w = result.w - surplus;
-        new_h = result.h * (gdouble) new_w / result.w;
-        result.x += surplus / 2;
-        result.y += (result.h - new_h) / 2;
-        result.w = new_w;
-        result.h = new_h;
-      } else {
-        /* cut off the surplus of width */
-        result.w -= surplus;
-        result.x += surplus / 2;
-      }
-    }
-#endif
 #else
     gst_video_sink_center_rect (src, dfbvideosink->window, &result, FALSE);
     result.x += dfbvideosink->window.x;
-- 
1.7.10.4

